paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/PaparazziViewOwners.kt

```diff
@@ -0,0 +1,48 @@
+package app.cash.paparazzi.internal
+
+import android.view.View
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LifecycleRegistry
+import androidx.lifecycle.ViewTreeLifecycleOwner
+import androidx.savedstate.SavedStateRegistry
+import androidx.savedstate.SavedStateRegistryController
+import androidx.savedstate.SavedStateRegistryOwner
+import androidx.savedstate.setViewTreeSavedStateRegistryOwner
+
+internal object PaparazziViewOwners {
+  private val hasLifecycleOwnerRuntime = isPresentInClassPath("androidx.lifecycle.LifecycleOwner")
+  private val hasSavedStateRegistryOwnerRuntime = isPresentInClassPath("androidx.savedstate.SavedStateRegistryOwner")
+
+  fun register(view: View) {
+    if (hasLifecycleOwnerRuntime) {
+      val lifecycleOwner = PaparazziLifecycleOwner()
+      ViewTreeLifecycleOwner.set(view, lifecycleOwner)
+
+      if (hasSavedStateRegistryOwnerRuntime) {
+        view.setViewTreeSavedStateRegistryOwner(PaparazziSavedStateRegistryOwner(lifecycleOwner))
+      }
+    }
+  }
+
+  private fun isPresentInClassPath(className: String): Boolean {
```

Comment:
```suggestion
  private fun isPresentInClasspath(className: String): Boolean {
```

Reply:
move this out to Paparazzi, since it's the consumer and therefore the classpath checker

Reply:
continuing this discussion in https://github.com/cashapp/paparazzi/pull/668#discussion_r1055912569 as I believe these two are inter-related.